### ISSUE TITLE
I3013: BackupRetentionPeriod incorrectly flagged for Aurora instance

### ISSUE THREAD

**Issue by jbarnes** [aws-cloudformation/cfn-lint#2319]
**cfn-lint version**: `cfn-lint 0.61.4`

This is in a similar vain to issues;

- https://github.com/aws-cloudformation/cfn-lint/issues/2313
- https://github.com/aws-cloudformation/cfn-lint/issues/2311

I3013: The default retention period will delete the data after a pre-defined time.

This is being flagged incorrectly for the `AWS::RDS::DBInstance` resource when it is associated with an Aurora cluster.

The `BackupRetentionPeriod` property is not required if the database instance is part of an Aurora cluster. 

I expect this is going to continue to flag new items for Aurora based database instances until all non required properties for an Aurora database instance are correctly ignored, I am simply raising the issue for the next cab off the rank in regards to the fix implemented https://github.com/aws-cloudformation/cfn-lint/pull/2317 which has since resolved the previous issues mentioned above.

**AWS Documentation reference** - https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-rds-database-instance.html#cfn-rds-dbinstance-backupretentionperiod

> Amazon Aurora
> Not applicable. The retention period for automated backups is managed by the DB cluster.

**CloudFormation Example**

```yaml
AuroraInstance:
    Type: AWS::RDS::DBInstance
    DeletionPolicy: Retain
    UpdateReplacePolicy: Retain
    Properties:
      AllowMajorVersionUpgrade: false
      AutoMinorVersionUpgrade: !Ref AutoMinorVersionUpgrade
      DBClusterIdentifier: !Ref AuroraCluster
      DBInstanceClass: !Ref InstanceClass
      DBInstanceIdentifier: "MyAuroraInstance"
      DBParameterGroupName: !Ref ParamGroup
      DBSubnetGroupName: !Ref SubnetGroup
      DeleteAutomatedBackups: !Ref DeleteAutomatedBackups
      EnablePerformanceInsights: !Ref EnablePerformanceInsights
      Engine: aurora-postgresql
      PerformanceInsightsKMSKeyId: !Ref KmsKey
      PerformanceInsightsRetentionPeriod: 7
      PubliclyAccessible: false
```

**Comment by jbarnes** [comment]
Rather than creating a new issue after every fix, in addition to the `BackupRetentionPeriod` property not being valid for Aurora database instances, the following other properties are also not required for the `AWS::RDS::DBInstance`.

It is **currently unknown** if the following are also incorrectly flagged under I3013, however the following are **not applicable** to an `AWS::RDS::DBInstance` that is part of an Aurora database cluster;

- `AllocatedStorage`
- `AssociatedRoles`
- `CharacterSetName`
- `CopyTagsToSnapshot`
- `DBName`
- `DBSnapshotIdentifier`
- `DBSubnetGroupName`
- `DeleteAutomatedBackups`
- `DeletionProtection`
- `DomainIAMRoleName`
- `EnableCloudwatchLogsExports`
- `EnableIAMDatabaseAuthentication`
- `EngineVersion`
- `KmsKeyId`
- `MasterUsername`
- `MasterUserPassword`
- `MaxAllocatedStorage`
- `MultiAZ`
- `Port`
- `PreferredBackupWindow`
- `ProcessorFeatures`
- `StorageEncrypted`
- `StorageType`
- `VPCSecurityGroups`

**Comment by PatMyron** [comment]
noting those issues were auto-closed since they're linked to the PR merged, but that PR hasn't been released to PyPI yet

**Comment by jbarnes** [comment]
> noting those issues were auto-closed since they're linked to the PR merged, but that PR hasn't been released to PyPI yet

Just realised I created this new issue for the same property 😓 , thought it was different to my original issue ticket. Can probably close this issue unless it's worth retaining the full list in the second comment.
